src/file-events/cpp/generic_fsnotifier.cpp

```diff
@@ -44,6 +44,14 @@ FileWatcherException::FileWatcherException(const string& message)
     : runtime_error(message) {
 }
 
+JavaExceptionThrownException::JavaExceptionThrownException()
+    : runtime_error("Java exception thrown from native code") {
+}
```

Comment:
Let's add some comments/docs here explaining what this is.

Reply:
I added some documentation in the header file.